FirebaseStorageUI/UIImageView+FirebaseStorage.h

```diff
@@ -0,0 +1,147 @@
+ *   could not be found in cache).
+ */
+- (nullable id<FUIDownloadTask>)sd_setImageWithStorageReference:(id<FUIStorageReference>)storageRef
+                                                placeholderImage:(nullable UIImage *)placeholder
+                                                      completion:(void (^_Nullable)(UIImage *_Nullable,
+                                                                                    NSError *_Nullable,
+                                                                                    SDImageCacheType,
+                                                                                    id<FUIStorageReference>))completion;
+
+/**
+ * Sets the image view's image to an image downloaded from the Firebase Storage reference.
+ *
+ * @param storageRef  A Firebase Storage reference containing an image.
+ * @param size        The maximum size of the downloaded image. If the downloaded image
+ *   exceeds this size, an error will be raised in the completion block.
+ * @param placeholder An image to display while the download is in progress.
+ * @param completion  A closure to handle events when the image finishes downloading.
+ *   The closure is not guaranteed to be invoked on the main thread.
+ * @return Returns a FIRStorageDownloadTask if a download was created (i.e. image
+ *   could not be found in cache).
+ */
+- (nullable id<FUIDownloadTask>)sd_setImageWithStorageReference:(id<FUIStorageReference>)storageRef
+                                                    maxImageSize:(UInt64)size
+                                                placeholderImage:(nullable UIImage *)placeholder
+                                                      completion:(void (^_Nullable)(UIImage *_Nullable,
+                                                                                    NSError *_Nullable,
+                                                                                    SDImageCacheType,
+                                                                                    id<FUIStorageReference>))completion;
+
+/**
+ * Sets the image view's image to an image downloaded from the Firebase Storage reference.
+ *
+ * @param storageRef  A Firebase Storage reference containing an image.
+ * @param size        The maximum size of the downloaded image. If the downloaded image
+ *   exceeds this size, an error will be raised in the completion block.
+ * @param placeholder An image to display while the download is in progress.
+ * @param cache       An image cache to check for images before downloading.
+ * @param completion  A closure to handle events when the image finishes downloading.
+ *   The closure is not guaranteed to be invoked on the main thread.
+ * @return Returns a FIRStorageDownloadTask if a download was created (i.e. image
+ *   could not be found in cache).
+ */
+- (nullable id<FUIDownloadTask>)sd_setImageWithStorageReference:(id<FUIStorageReference>)storageRef
```

Comment:
These aren't initializer methods, though a designator of some sort would be cool